src/controls/trackpad-controls.js

```diff
@@ -59,24 +60,63 @@ module.exports = AFRAME.registerComponent('trackpad-controls', {
   },
 
   onTouchStart: function (e) {
-    this.isMoving = true;
+    this.startingAxisData = [];
+    this.canRecordAxis = true
     e.preventDefault();
   },
 
   onTouchEnd: function (e) {
+    this.startingAxisData = []
     this.isMoving = false;
     e.preventDefault();
   },
 
   onAxisMove: function(e){
     var axis_data = e.detail.axis;
 
-    if(axis_data[1] < 0){
-      this.zVel = 1;
+    if(this.canRecordAxis){
+      this.canRecordAxis = false
+
+      this.startingAxisData[0] = axis_data[0]
+      this.startingAxisData[1] = axis_data[1]
     }
 
-    if(axis_data[1] > 0){
-      this.zVel = -1;
+
+    if(this.startingAxisData.length > 0){
+      var movingLeft     = this.startingAxisData[0] > 0 && axis_data[0] < 0
+      var movingRight    = this.startingAxisData[0] < 0 && axis_data[0] > 0
+      var movingForward  = this.startingAxisData[1] > 0 && axis_data[1] < 0
+      var movingBackward = this.startingAxisData[1] < 0 && axis_data[1] > 0
```

Comment:
I'm not sure I understand this block; why must the starting axis data be > 0 for the player to be moving left? Shouldn't it be enough that `axis_data[0] < startingAxisData[0]`? The swipe cannot begin in the middle of the trackpad? It looks to me like this code block shouldn't need to handle left/right/forward/back as four separate cases, but just two cases for Z and X axes.